airframe-sql/src/test/scala/wvlet/airframe/sql/analyzer/TypeResolverTest.scala

```diff
@@ -558,4 +548,54 @@ class TypeResolverTest extends AirSpec {
       }
     }
   }
+
+  test("resolve count(*)") {
+    test("resolve simple count(*)") {
+      val p = analyze("select count(*) from A")
+      p.outputAttributes match {
+        case List(SingleColumn(FunctionCall("count", Seq(c @ AllColumns(_, _, _)), _, _, _, _), _, _, _)) =>
+          c.sourceTables shouldBe Some(Seq(tableA))
+        case _ => fail(s"unexpected plan:\n${p.pp}")
+      }
+    }
+
+    test("resolve count(*) in expression") {
+      val p = analyze("select count(*) + 1 from A")
+      p.outputAttributes match {
+        case List(
+              SingleColumn(
+                ArithmeticBinaryExpr(
+                  _,
+                  FunctionCall("count", Seq(c @ AllColumns(_, _, _)), _, _, _, _),
+                  LongLiteral(1, _),
+                  _
+                ),
+                _,
+                _,
+                _
+              )
+            ) =>
+          c.sourceTables shouldBe Some(Seq(tableA))
+        case _ => fail(s"unexpected plan:\n${p.pp}")
+      }
+    }
+
+    test("resolve count(*) in sub query") {
+      val p = analyze("select cnt from (select count(*) as cnt from A)")
+      p.outputAttributes match {
+        case List(SingleColumn(FunctionCall("count", Seq(c @ AllColumns(_, _, _)), _, _, _, _), _, _, _)) =>
+          c.sourceTables shouldBe Some(Seq(tableA))
+        case _ => fail(s"unexpected plan:\n${p.pp}")
+      }
+    }
+
+    test("resolve count(*) in CTE") {
+      val p = analyze("WITH q AS (select count(*) as cnt from A) select cnt from q")
+      p.outputAttributes match {
+        case List(SingleColumn(FunctionCall("count", Seq(c @ AllColumns(_, _, _)), _, _, _, _), _, _, _)) =>
```

Comment:
haha. Embedding case class parameters with `_` becoming scary 😅  Future test cases should use type reference like `f:FunctionCall if f.name == "count"`